plugin/kubernetes/pod.go

```diff
@@ -19,56 +19,98 @@ import (
 
 type pod struct {
 	plugin.EntryBase
-	client    *k8s.Clientset
-	config    *rest.Config
-	ns        string
-	startTime time.Time
+	client *k8s.Clientset
+	config *rest.Config
+	ns     string
 }
 
-func newPod(client *k8s.Clientset, config *rest.Config, ns string, p *corev1.Pod) *pod {
+func newPod(ctx context.Context, client *k8s.Clientset, config *rest.Config, ns string, p *corev1.Pod) (*pod, error) {
 	pd := &pod{
 		EntryBase: plugin.NewEntry(p.Name),
 		client:    client,
 		config:    config,
 		ns:        ns,
-		startTime: p.CreationTimestamp.Time,
 	}
+	pd.DisableDefaultCaching()
 
-	return pd
-}
-
-func (p *pod) Metadata(ctx context.Context) (plugin.MetadataMap, error) {
-	pd, err := p.client.CoreV1().Pods(p.ns).Get(p.Name(), metav1.GetOptions{})
+	pdInfo, err := pd.cachedPodInfo(ctx, false)
 	if err != nil {
 		return nil, err
 	}
+	pdInfo.pd = p
+
+	meta := pdInfo.toMeta()
+	attr := plugin.EntryAttributes{}
+	attr.SetCtime(p.CreationTimestamp.Time)
+	attr.SetAtime(attr.Ctime())
+	attr.SetMeta(meta)
```

Comment:
This doesn't set Size anywhere (from `meta`) so it's not available when you list it.

Reply:
Updated.